pkg/controller/operconfig/operconfig_controller.go

```diff
@@ -271,21 +275,38 @@ func (r *ReconcileOperConfig) Reconcile(ctx context.Context, request reconcile.R
 	}
 	objs = append([]*uns.Unstructured{app}, objs...)
 
-	// Set up the Pod reconciler before we start creating DaemonSets/Deployments
-	daemonSets := []types.NamespacedName{}
-	deployments := []types.NamespacedName{}
+	// Set up the Pod reconciler before we start creating DaemonSets/Deployments/StatefulSets
+	daemonSets := []statusmanager.ClusteredName{}
+	deployments := []statusmanager.ClusteredName{}
+	statefulSets := []statusmanager.ClusteredName{}
 	relatedObjects := []configv1.ObjectReference{}
+	relatedClusterObjects := []network.RelatedObject{}
 	for _, obj := range objs {
 		if obj.GetAPIVersion() == "apps/v1" && obj.GetKind() == "DaemonSet" {
-			daemonSets = append(daemonSets, types.NamespacedName{Namespace: obj.GetNamespace(), Name: obj.GetName()})
+			daemonSets = append(daemonSets, statusmanager.ClusteredName{ClusterName: obj.GetClusterName(), Namespace: obj.GetNamespace(), Name: obj.GetName()})
```

Comment:
Will the cluster name be `""` or `"default"` for most objects? And does the rest of the code reflect that?

Reply:
aha, ClusterClient treats `""` as default.

Reply:
I think It will be the default value `""` for most objects. and `""` is considered as `default`. The rest of code use `ClientFor(cluster name)` to access the right client.